src/prop/minisat/simp/SimpSolver.cc

```diff
@@ -62,27 +61,18 @@ SimpSolver::SimpSolver(Env& env,
       use_asymm(opt_use_asymm),
       // make sure this is not enabled if unsat cores or proofs are on
       use_rcheck(opt_use_rcheck && !options().smt.unsatCores && !pnm),
-      use_elim(options().prop.minisatUseElim && !enableIncremental),
       merges(0),
       asymm_lits(0),
       eliminated_vars(0),
       elimorder(1),
-      use_simplification(!enableIncremental && !options().smt.unsatCores
-                         && !pnm)  // TODO: turn off simplifications if
-                                   // proofs are on initially
-      ,
+      use_simplification(
```

Comment:
We should not be passing `pnm` to SimpSolver, nor checking whether it is non-null here. I think we need to change `!pnm` in this check to `!env.isSatProofProducing()`.

Reply:
Ping.

Would also help to make this not check for incremental here.

Reply:
This requires changing several interfaces, I think it's better to be done in anather PR.

Reply:
Ok